src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs

```diff
@@ -167,44 +154,44 @@ public class SubscriptionManager
         // ReSharper disable once UnusedMember.Global
         public async Task<IActionResult> PublicationUnsubscribeFunc(
             [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "publication/{id}/unsubscribe/{token}")]
-            HttpRequest req,
             FunctionContext context,
             string id,
             string token)
         {
-            _logger.LogInformation("{FunctionName} triggered", context.FunctionDefinition.Name);
+            logger.LogInformation("{FunctionName} triggered", context.FunctionDefinition.Name);
 
             var email = _tokenService.GetEmailFromToken(token);
-
-            if (email != null)
+            if (email is null)
             {
-                var table = await _storageTableService.GetTable(NotifierSubscriptionsTableName);
-
-                var sub = new SubscriptionEntity(id, email);
-                sub = _storageTableService.RetrieveSubscriber(table, sub).Result;
+                return new BadRequestObjectResult("Unable to unsubscribe. A valid email address could not be parsed from the given token.");
+            }
 
-                if (sub != null)
-                {
-                    await _storageTableService.RemoveSubscriber(table, sub);
-                    return new RedirectResult(
-                        $"{_appSettingOptions.PublicAppUrl}/subscriptions?slug={sub.Slug}&unsubscribed=true",
-                        true);
-                }
+            var table = await _storageTableService.GetTable(NotifierSubscriptionsTableName);
+            var sub = _storageTableService.RetrieveSubscriber(table, new SubscriptionEntity(id, email)).Result;
```

Comment:
I know this isn't you, but I'd switch this `.Result` out for an `await`.

Reply:
Ooh good spot, yeah will change